src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,368 @@
+        end
+        it = iterate(_heap, it[2])
+    end
+    lo.rootList = removeMinFromTreeReturnBHeap(temp)
+    new_heap.nodecount = heap.nodecount - 1
+    union!(new_heap, lo, false)
+    adjust(new_heap)
+    heap.rootList = new_heap.rootList
+    heap.nodecount -= 1
+    return temp.data
+end
+
+function findhelper(node::Union{MutableBinomialHeapNode{T},Nothing}, i::Int) where T
+    if node === nothing 
+        return nothing
+    end
+    if node.handle == i 
+        return node
+    end
+    res = findhelper(node.child, i)
+    if res !== nothing 
+        return res
+    end
+    return findhelper(node.sibling, i)
+end
+    
+function findNode(heap::MutableBinomialHeap{T}, i::Int) where T 
+    _heap = heap.rootList
+    node = nothing
+    n = length(_heap)
+    if n == 0
+        return nothing
+    end
+    for j = 1:n
+        node = findhelper(_heap[j], i)
+        if node !== nothing
+            break
+        end
+    end
+    if node === nothing
+        throw(HeapBoundsError("attempt to access $(heap.nodecount) MutableBinomialHeap{$(T)} at index [$(heap.nodecount + 1)]"))
+    end
+    return node
+end
+function increaseKey!(node::MutableBinomialHeapNode{T}, new_val::T) where T
+    node.data = new_val
+    child = node.child
+
+    while child !== nothing && node.data > child.data
+        node.data, child.data = child.data, node.data
+        node.handle, child.handle = child.handle, node.handle
+        node = child
+        child = child.child
+    end
+end
+function decreaseKey!(heap::MutableBinomialHeap{T}, i::Int, new_val) where T
+    new_val = convert(T, new_val)
+    node = findNode(heap, i)
+    if node === nothing
+        return 
+    end
+    if node.data > new_val
+        node.data = new_val
+        parent = node.parent
+        while parent !== nothing && node.data < parent.data
+            node.data, parent.data = parent.data, node.data
+            node.handle, parent.handle = parent.handle, node.handle
+            node = parent
+            parent = parent.parent
+        end
+    else increaseKey!(node, new_val)
+    end
+end
+
+function top_elem_with_handle(heap::MutableBinomialHeap{T}) where T 
+    _heap = heap.rootList
+    if length(_heap) == 0
+        return nothing
+    end
+    it = iterate(_heap)
+    @inbounds temp = _heap[1]
+    while it !== nothing
+        elem, state = it
+        if elem.data < temp.data 
+            temp = elem 
+        end
+        it = iterate(_heap, state) 
+    end
+    return (temp.data, temp)
+end
+
+function update_handle(c::Int, h::MutableBinomialHeap)
+    node::Union{Nothing,MutableBinomialHeapNode{T}} where T = nothing 
+    for i = 1:h.cumm_nodecount
+        node = nothing
+        try
+            node = findNode(h, i)
+            node.handle = node.handle + c
+        catch _
+            continue
+        end
+    end
+end
+
+function union!(h1::MutableBinomialHeap{T}, h2::MutableBinomialHeap{T}, flag::Bool) where T
```

Comment:
I think it would be nice to write what `flag` does, either in the name (i.e. call it `update_handle` or something), or in a comment above the function. Perhaps more importantly, consider renaming this function. `union!` is a `Base` function, but this signature does not match the signature expected of `Base.union!`. Perhaps just call it `_union!`.